packages/kbn-esql-validation-autocomplete/src/definitions/builtin.ts

```diff
@@ -130,39 +121,119 @@ function createComparisonDefinition(
   };
 }
 
+const addTypeTable: MathFunctionSignature[] = [
+  ['date_period', 'date_period', 'date_period'],
+  ['date_period', 'date', 'date'],
+  ['date', 'date_period', 'date'],
+  ['date', 'time_duration', 'date'],
+  ['date', 'time_literal', 'date'],
+  ['double', 'double', 'double'],
+  ['double', 'integer', 'double'],
+  ['double', 'long', 'double'],
+  ['integer', 'double', 'double'],
+  ['integer', 'integer', 'integer'],
+  ['integer', 'long', 'long'],
+  ['long', 'double', 'double'],
+  ['long', 'integer', 'long'],
+  ['long', 'long', 'long'],
+  ['time_duration', 'date', 'date'],
+  ['time_duration', 'time_duration', 'time_duration'],
+  ['unsigned_long', 'unsigned_long', 'unsigned_long'],
+  ['time_literal', 'date', 'date'],
+];
+
+const subtractTypeTable: MathFunctionSignature[] = [
+  ['date_period', 'date_period', 'date_period'],
+  ['date', 'date_period', 'date'],
+  ['date', 'time_duration', 'date'],
+  ['date', 'time_literal', 'date'],
+  ['double', 'double', 'double'],
+  ['double', 'integer', 'double'],
+  ['double', 'long', 'double'],
+  ['integer', 'double', 'double'],
+  ['integer', 'integer', 'integer'],
+  ['integer', 'long', 'long'],
+  ['long', 'double', 'double'],
+  ['long', 'integer', 'long'],
+  ['long', 'long', 'long'],
+  ['time_duration', 'date', 'date'],
+  ['time_duration', 'time_duration', 'time_duration'],
+  ['unsigned_long', 'unsigned_long', 'unsigned_long'],
+  ['time_literal', 'date', 'date'],
+];
+
+const multiplyTypeTable: MathFunctionSignature[] = [
+  ['double', 'double', 'double'],
+  ['double', 'integer', 'double'],
+  ['double', 'long', 'double'],
+  ['integer', 'double', 'double'],
+  ['integer', 'integer', 'integer'],
+  ['integer', 'long', 'long'],
+  ['long', 'double', 'double'],
+  ['long', 'integer', 'long'],
+  ['long', 'long', 'long'],
+  ['unsigned_long', 'unsigned_long', 'unsigned_long'],
+];
+
+const divideTypeTable: MathFunctionSignature[] = [
+  ['double', 'double', 'double'],
+  ['double', 'integer', 'double'],
+  ['double', 'long', 'double'],
+  ['integer', 'double', 'double'],
+  ['integer', 'integer', 'integer'],
+  ['integer', 'long', 'long'],
+  ['long', 'double', 'double'],
+  ['long', 'integer', 'long'],
+  ['long', 'long', 'long'],
+  ['unsigned_long', 'unsigned_long', 'unsigned_long'],
+];
+
+const modulusTypeTable: MathFunctionSignature[] = [
+  ['double', 'double', 'double'],
+  ['double', 'integer', 'double'],
+  ['double', 'long', 'double'],
+  ['integer', 'double', 'double'],
+  ['integer', 'integer', 'integer'],
+  ['integer', 'long', 'long'],
+  ['long', 'double', 'double'],
+  ['long', 'integer', 'long'],
+  ['long', 'long', 'long'],
+  ['unsigned_long', 'unsigned_long', 'unsigned_long'],
+];
```

Comment:
Curious: how did you work all these out?

Reply:
From the Elasticsearch documentation https://www.elastic.co/guide/en/elasticsearch/reference/master/esql-functions-operators.html#_add